app/web/features/communities/events/DiscoverEventsList.test.tsx

```diff
@@ -0,0 +1,271 @@
+    value: GeocodeResult;
+    onChange: (newLocationResult: GeocodeResult) => void;
+  }> = (props) => (
+    <input
+      data-testid="location-autocomplete"
+      value={props.value?.name}
+      onChange={(e) => {
+        props.onChange({
+          simplifiedName: e.target.value,
+          name: e.target.value,
+          location: new LngLat(0, 0),
+          bbox: [0, 0, 0, 0],
+        });
+      }}
+    />
+  );
+  LocationAutocomplete.displayName = "LocationAutocomplete";
+  return LocationAutocomplete;
+});
+
+describe("DiscoverEventsList", () => {
+  const mockUseTranslation = useTranslation as jest.Mock;
+  const mockUseForm = useForm as jest.Mock;
+  const mockUseEventSearch = useEventSearch as jest.Mock;
+  const mockUseAuthContext = useAuthContext as jest.Mock;
+  const mockUseController = useController as jest.Mock;
+  const mockPagination = Pagination as jest.Mock;
+
+  beforeEach(() => {
+    mockUseTranslation.mockReturnValue({
+      t: (key: string) => key,
+    });
+    mockUseForm.mockReturnValue({
+      control: {},
+      errors: {},
+    });
+    mockUseController.mockReturnValue({
+      field: {
+        onChange: jest.fn(),
+        onBlur: jest.fn(),
+        value: "",
+      },
+    });
+
+    // Mock the return value of useAuthContext to include authState with userId
+    mockUseAuthContext.mockReturnValue({
+      authState: { authenticated: true, userId: 1 },
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Renders the component correctly with empty state", () => {
+    mockUseEventSearch.mockReturnValue({
+      data: null,
+      error: null,
+      isLoading: false,
+    });
+
+    render(<DiscoverEventsList />);
+
+    expect(
+      screen.getByText("communities:discover_events_title")
+    ).toBeInTheDocument();
+    expect(screen.getByText("communities:communities")).toBeInTheDocument();
+    expect(screen.getByText("communities:online")).toBeInTheDocument();
+    expect(screen.getByTestId("location-autocomplete")).toBeInTheDocument();
+    expect(
+      screen.getByText("communities:events_empty_state")
+    ).toBeInTheDocument();
+  });
+
+  it("Renders loading state", () => {
+    mockUseEventSearch.mockReturnValue({
+      data: null,
+      error: null,
+      isLoading: true,
+    });
+
+    render(<DiscoverEventsList />);
+
+    expect(screen.getByRole("progressbar")).toBeInTheDocument();
+  });
+
+  it("Renders error message when there is an error", () => {
+    mockUseEventSearch.mockReturnValue({
+      data: null,
+      error: { message: "Error occurred" },
+      isLoading: false,
+    });
+
+    render(<DiscoverEventsList />);
+
+    expect(screen.getByText("Error occurred")).toBeInTheDocument();
+  });
+
+  it("Renders events and pagination when data is available", () => {
+    mockUseEventSearch.mockReturnValue({
+      data: {
+        eventsList: mockEvents.filter((event) => !event.isCancelled),
+        totalItems: 4,
```

Comment:
pagination with 4 items? (I guess it's disabled?)

Reply:
It's just for the test so I don't need to make s much mock data.